File: src/codegen/io.py
from ..utils.nodes import *
from .utils import *


LIB_NAME = "io"

IO_SYMBOL_LIST = [
    # Integer I/O
    Symbol("readInt", FunctionType([], PrimitiveType("int")), CName(LIB_NAME)),
    Symbol("writeInt", FunctionType([PrimitiveType("int")], PrimitiveType("void")), CName(LIB_NAME)),
    Symbol("writeIntLn", FunctionType([PrimitiveType("int")], PrimitiveType("void")), CName(LIB_NAME)),
    
    # Float I/O
    Symbol("readFloat", FunctionType([], PrimitiveType("float")), CName(LIB_NAME)),
    Symbol("writeFloat", FunctionType([PrimitiveType("float")], PrimitiveType("void")), CName(LIB_NAME)),
    Symbol("writeFloatLn", FunctionType([PrimitiveType("float")], PrimitiveType("void")), CName(LIB_NAME)),
    
    # Boolean I/O
    Symbol("readBool", FunctionType([], PrimitiveType("boolean")), CName(LIB_NAME)),
    Symbol("writeBool", FunctionType([PrimitiveType("boolean")], PrimitiveType("void")), CName(LIB_NAME)),
    Symbol("writeBoolLn", FunctionType([PrimitiveType("boolean")], PrimitiveType("void")), CName(LIB_NAME)),
    
    # String I/O
    Symbol("readStr", FunctionType([], PrimitiveType("string")), CName(LIB_NAME)),
    Symbol("writeStr", FunctionType([PrimitiveType("string")], PrimitiveType("void")), CName(LIB_NAME)),
    Symbol("writeStrLn", FunctionType([PrimitiveType("string")], PrimitiveType("void")), CName(LIB_NAME)),
]

File: src/runtime/io.java
import java.util.Scanner;

/**
 * IO class for OPLang runtime.
 * Provides input/output operations for OPLang programs.
 */
public class io {
    private static Scanner scanner = new Scanner(System.in);
    
    // Integer I/O
    public static int readInt() {
        return scanner.nextInt();
    }
    
    public static void writeInt(int anArg) {
        System.out.print(anArg);
    }
    
    public static void writeIntLn(int anArg) {
        System.out.println(anArg);
    }
    
    // Float I/O
    public static float readFloat() {
        return scanner.nextFloat();
    }
    
    public static void writeFloat(float anArg) {
        System.out.print(anArg);
    }
    
    public static void writeFloatLn(float anArg) {
        System.out.println(anArg);
    }
    
    // Boolean I/O
    public static boolean readBool() {
        String input = scanner.next().toLowerCase();
        return input.equals("true") || input.equals("1");
    }
    
    public static void writeBool(boolean anArg) {
        System.out.print(anArg);
    }
    
    public static void writeBoolLn(boolean anArg) {
        System.out.println(anArg);
    }
    
    // String I/O
    public static String readStr() {
        return scanner.next();
    }
    
    public static void writeStr(String anArg) {
        System.out.print(anArg);
    }
    
    public static void writeStrLn(String anArg) {
        System.out.println(anArg);
    }
}

